Terraform crash when planning a new resource containing a `SingleNestedAttribute`

<!--
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

If your issue relates to Terraform Cloud/Enterprise, please contact tf-cloud@hashicorp.support.
If your issue relates to a specific Terraform provider, please open it in the provider's own repository. The index of providers is at https://registry.terraform.io/browse/providers.

To fix problems, we need clear reproduction cases - we need to be able to see it happen locally. A reproduction case is ideally something a Terraform Core engineer can git-clone or copy-paste and run immediately, without inventing any details or context. 

* A short example can be directly copy-pasteable; longer examples should be in separate git repositories, especially if multiple files are needed
* Please include all needed context. For example, if you figured out that an expression can cause a crash, put the expression in a variable definition or a resource
* Set defaults on (or omit) any variables. The person reproducing it should not need to invent variable settings
* If multiple steps are required, such as running terraform twice, consider scripting it in a simple shell script. Providing a script can be easier than explaining what changes to make to the config between runs.
* Omit any unneeded complexity: remove variables, conditional statements, functions, modules, providers, and resources that are not needed to trigger the bug
* When possible, use the [null resource](https://www.terraform.io/docs/providers/null/resource.html) provider rather than a real provider in order to minimize external dependencies. We know this isn't always feasible. The Terraform Core team doesn't have deep domain knowledge in every provider, or access to every cloud platform for reproduction cases.

-->

### Terraform Version
<!---
Run `terraform version` to show the version, and paste the result between the ``` marks below.

If you are not running the latest version of Terraform, please try upgrading because your issue may have already been fixed.
-->

```console
% terraform version
Terraform v1.0.4
on darwin_amd64
```

### Terraform Configuration Files
<!--
Paste the relevant parts of your Terraform configuration between the ``` marks below.

For Terraform configs larger than a few resources, or that involve multiple files, please make a GitHub repository that we can clone, rather than copy-pasting multiple files in here. For security, you can also encrypt the files using our GPG public key at https://www.hashicorp.com/security.
-->

```terraform
resource "aws_xray_group" "test" {}
```

### Debug Output
<!--
Full debug output can be obtained by running Terraform with the environment variable `TF_LOG=trace`. Please create a GitHub Gist containing the debug output. Please do _not_ paste the debug output in the issue, since debug output is long.

Debug output may contain sensitive information. Please review it before posting publicly, and if you are concerned feel free to encrypt the files using the HashiCorp security public key.
-->

### Crash Output
<!--
If the console output indicates that Terraform crashed, please share a link to a GitHub Gist containing the output of the `crash.log` file.
-->

```
% terraform plan
╷
│ Warning: Provider development overrides are in effect
│ 
│ The following provider development overrides are set in the CLI configuration:
│  - hashicorp/aws-xxxxxxxx in /Users/ewbankkit/go/bin
│  - hashicorp/aws in /Users/ewbankkit/go/bin
│ 
│ The behavior may therefore not match any released version of the provider and applying changes may cause the state to become incompatible with published releases.
╵
panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 102 [running]:
github.com/zclconf/go-cty/cty.Value.GetAttr(0x37490f8, 0xc000d7ee00, 0x0, 0x0, 0xc0005a5ad0, 0x10, 0x3749018, 0xc0001a836b, 0x0, 0x0)
	/Users/distiller/go/pkg/mod/github.com/zclconf/go-cty@v1.9.0/cty/value_ops.go:760 +0x416
github.com/hashicorp/terraform/internal/plans/objchange.proposedNewAttributes(0xc00118b830, 0x37490f8, 0xc000d23bc0, 0x2f418e0, 0xc0006247b0, 0x37490f8, 0xc000d7ee00, 0x0, 0x0, 0x1)
	/Users/distiller/project/project/internal/plans/objchange/objchange.go:267 +0x26f
github.com/hashicorp/terraform/internal/plans/objchange.proposedNewNestedType(0xc000ac6420, 0x37490f8, 0xc000d23ba0, 0x0, 0x0, 0x37490f8, 0xc000d7ee00, 0x0, 0x0, 0x0, ...)
	/Users/distiller/project/project/internal/plans/objchange/objchange.go:314 +0x813
github.com/hashicorp/terraform/internal/plans/objchange.proposedNewAttributes(0xc00118b6e0, 0x37490f8, 0xc000d23bb0, 0x2f418e0, 0xc0006244e0, 0x37490f8, 0xc000d23b80, 0x2f418e0, 0xc000624180, 0xc0001a836a)
	/Users/distiller/project/project/internal/plans/objchange/objchange.go:295 +0x49a
github.com/hashicorp/terraform/internal/plans/objchange.proposedNew(0xc00118b740, 0x37490f8, 0xc000d23bb0, 0x2f418e0, 0xc0006244e0, 0x37490f8, 0xc000d23b80, 0x2f418e0, 0xc000624180, 0x20, ...)
	/Users/distiller/project/project/internal/plans/objchange/objchange.go:80 +0x165
github.com/hashicorp/terraform/internal/plans/objchange.ProposedNew(0xc00118b740, 0x37490f8, 0xc000d236d0, 0x0, 0x0, 0x37490f8, 0xc000d23b80, 0x2f418e0, 0xc000624180, 0x0, ...)
	/Users/distiller/project/project/internal/plans/objchange/objchange.go:42 +0x11f
github.com/hashicorp/terraform/internal/terraform.(*NodeAbstractResourceInstance).plan(0xc000034380, 0x3779820, 0xc0001508c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/distiller/project/project/internal/terraform/node_resource_abstract_instance.go:732 +0xdab
github.com/hashicorp/terraform/internal/terraform.(*NodePlannableResourceInstance).managedResourceExecute(0xc001226840, 0x3779820, 0xc0001508c0, 0x100c408, 0x32a3600, 0xc001721c50)
	/Users/distiller/project/project/internal/terraform/node_resource_plan_instance.go:196 +0x8cb
github.com/hashicorp/terraform/internal/terraform.(*NodePlannableResourceInstance).Execute(0xc001226840, 0x3779820, 0xc0001508c0, 0x6710902, 0xc001721ce8, 0x100d305, 0x2f8ae40)
	/Users/distiller/project/project/internal/terraform/node_resource_plan_instance.go:54 +0xb3
github.com/hashicorp/terraform/internal/terraform.(*ContextGraphWalker).Execute(0xc0016c0410, 0x3779820, 0xc0001508c0, 0x6710920, 0xc001226840, 0x0, 0x0, 0x0)
	/Users/distiller/project/project/internal/terraform/graph_walk_context.go:129 +0xbf
github.com/hashicorp/terraform/internal/terraform.(*Graph).walk.func1(0x32a3600, 0xc001226840, 0x0, 0x0, 0x0)
	/Users/distiller/project/project/internal/terraform/graph.go:59 +0xbd3
github.com/hashicorp/terraform/internal/dag.(*Walker).walkVertex(0xc0015c6300, 0x32a3600, 0xc001226840, 0xc0015bc3c0)
	/Users/distiller/project/project/internal/dag/walk.go:381 +0x288
created by github.com/hashicorp/terraform/internal/dag.(*Walker).Update
	/Users/distiller/project/project/internal/dag/walk.go:304 +0x1246




!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.688954863.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

SECURITY WARNING: the "crash.688954863.log" file that was created may contain 
sensitive information that must be redacted before it is safe to share 
on the issue tracker.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

[crash.688954863.log](https://github.com/hashicorp/terraform/files/6994168/crash.688954863.log)

### Expected Behavior
<!--
What should have happened?
-->

### Actual Behavior
<!--
What actually happened?
-->

### Steps to Reproduce
<!--
Please list the full steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform apply`
-->

### Additional Context
<!--
Are there anything atypical about your situation that we should know? For example: is Terraform running in a wrapper script or in a CI system? Are you passing any unusual command line options or environment variables to opt-in to non-default behavior?
-->

**N.B.** This is not the [usual AWS Provider](https://github.com/hashicorp/terraform-provider-aws) but is an under-development version using https://github.com/hashicorp/terraform-plugin-framework (and hence protocol v6).

The resource's schema includes a `SingleNestedAttribute`:

```go
		"insights_configuration": {
			Attributes: schema.SingleNestedAttributes(
				map[string]schema.Attribute{
					"insights_enabled": {
						// Property: InsightsEnabled
						Description: "Set the InsightsEnabled value to true to enable insights or false to disable insights.",
						Type:        types.BoolType,
						Optional:    true,
					},
					"notifications_enabled": {
						// Property: NotificationsEnabled
						Description: "Set the NotificationsEnabled value to true to enable insights notifications. Notifications can only be enabled on a group with InsightsEnabled set to true.",
						Type:        types.BoolType,
						Optional:    true,
					},
				},
			),
			Optional: true,
		},
```

### References
<!--
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:

- #6017

-->

--- a/internal/plans/objchange/objchange.go
+++ b/internal/plans/objchange/objchange.go
@@ -311,8 +311,11 @@ func proposedNewNestedType(schema *configschema.Object, prior, config cty.Value)
 	var newV cty.Value
 	switch schema.Nesting {
 	case configschema.NestingSingle:
-		newAttrs := proposedNewAttributes(schema.Attributes, prior, config)
-		newV = cty.ObjectVal(newAttrs)
+		if !config.IsNull() {
+			newV = cty.ObjectVal(proposedNewAttributes(schema.Attributes, prior, config))
+		} else {
+			newV = cty.NullVal(config.Type())
+		}
 
 	case configschema.NestingList:
 		// Nested blocks are correlated by index.

--- a/internal/plans/objchange/objchange_test.go
+++ b/internal/plans/objchange/objchange_test.go
@@ -1433,7 +1433,7 @@ func TestProposedNew(t *testing.T) {
 				}),
 			}),
 			cty.ObjectVal(map[string]cty.Value{
-				"single": cty.ObjectVal(map[string]cty.Value{"bar": cty.NullVal(cty.String)}),
+				"single": cty.NullVal(cty.Object(map[string]cty.Type{"bar": cty.String})),
 				"list":   cty.NullVal(cty.List(cty.Object(map[string]cty.Type{"bar": cty.String}))),
 				"map":    cty.NullVal(cty.Map(cty.Object(map[string]cty.Type{"bar": cty.String}))),
 				"set":    cty.NullVal(cty.Set(cty.Object(map[string]cty.Type{"bar": cty.String}))),
@@ -1447,7 +1447,7 @@ func TestProposedNew(t *testing.T) {
 				}))),
 			}),
 			cty.ObjectVal(map[string]cty.Value{
-				"single": cty.ObjectVal(map[string]cty.Value{"bar": cty.NullVal(cty.String)}),
+				"single": cty.NullVal(cty.Object(map[string]cty.Type{"bar": cty.String})),
 				"list":   cty.NullVal(cty.List(cty.Object(map[string]cty.Type{"bar": cty.String}))),
 				"map":    cty.NullVal(cty.Map(cty.Object(map[string]cty.Type{"bar": cty.String}))),
 				"set":    cty.NullVal(cty.Set(cty.Object(map[string]cty.Type{"bar": cty.String}))),

